### DISCUSSION TITLE
Help with config

### DISCUSSION THREAD

**Discussion by DartMitai** [saghen/blink.cmp#477] [Q&A] [answered]
plugins.lua
```
-- BLINK.CMP
    {
        'saghen/blink.cmp',
        lazy = false,
        dependencies = 'rafamadriz/friendly-snippets',
        version = 'v0.*',
        config = function()
            require('plugins.blink')
        end
    },
```
blink.lua
```
local blink = require('blink')
blink.setup{
   opts = {
    -- 'default' for mappings similar to built-in completion
    -- 'super-tab' for mappings similar to vscode (tab to accept, arrow keys to navigate)
    -- 'enter' for mappings similar to 'super-tab' but with 'enter' to accept
    -- see the "default configuration" section below for full documentation on how to define
    -- your own keymap.
    keymap = { preset = 'default' },

    appearance = {
      -- Sets the fallback highlight groups to nvim-cmp's highlight groups
      -- Useful for when your theme doesn't support blink.cmp
      -- will be removed in a future release
      use_nvim_cmp_as_default = true,
      -- Set to 'mono' for 'Nerd Font Mono' or 'normal' for 'Nerd Font'
      -- Adjusts spacing to ensure icons are aligned
      nerd_font_variant = 'mono'
    },

    -- default list of enabled providers defined so that you can extend it
    -- elsewhere in your config, without redefining it, via `opts_extend`
    sources = {
      completion = {
        enabled_providers = { 'lsp', 'path', 'snippets', 'buffer' },
      },
    },

    -- experimental auto-brackets support
    -- completion = { accept = { auto_brackets = { enabled = true } } }

    -- experimental signature help support
    -- signature = { enabled = true }
  },
  -- allows extending the enabled_providers array elsewhere in your config
  -- without having to redefine it
  opts_extend = { "sources.completion.enabled_providers" },
}
```
when I download, I get the following
![image](https://github.com/user-attachments/assets/4fc79605-71d8-4830-9499-757f4588ef14)

**Comment by saghen** [comment]
Hey there, put your options in the `opts` field of the lazy.nvim definition, and don't make any other calls. Internally, it will make a call like `require('blink.cmp').setup(your_opts)`.

```lua
{
  'saghen/blink.cmp',
  lazy = false,
  dependencies = 'rafamadriz/friendly-snippets',
  version = 'v0.*',
  opts = {
    -- 'default' for mappings similar to built-in completion
    -- 'super-tab' for mappings similar to vscode (tab to accept, arrow keys to navigate)
    -- 'enter' for mappings similar to 'super-tab' but with 'enter' to accept
    -- see the "default configuration" section below for full documentation on how to define
    -- your own keymap.
    keymap = { preset = 'default' },

    appearance = {
      -- Sets the fallback highlight groups to nvim-cmp's highlight groups
      -- Useful for when your theme doesn't support blink.cmp
      -- will be removed in a future release
      use_nvim_cmp_as_default = true,
      -- Set to 'mono' for 'Nerd Font Mono' or 'normal' for 'Nerd Font'
      -- Adjusts spacing to ensure icons are aligned
      nerd_font_variant = 'mono'
    },

    -- default list of enabled providers defined so that you can extend it
    -- elsewhere in your config, without redefining it, via `opts_extend`
    sources = {
      completion = {
        enabled_providers = { 'lsp', 'path', 'snippets', 'buffer' },
      },
    },

    -- experimental auto-brackets support
    -- completion = { accept = { auto_brackets = { enabled = true } } }

    -- experimental signature help support
    -- signature = { enabled = true }
  },
  -- allows extending the enabled_providers array elsewhere in your config
  -- without having to redefine it
  opts_extend = { "sources.completion.enabled_providers" },
}

**Reply by DartMitai** [reply]
![image](https://github.com/user-attachments/assets/dad4c783-a05d-4456-ba7e-addee4359aae)

**Reply by DartMitai** [reply]
> Hey there, put your options in the `opts` field of the lazy.nvim definition, and don't make any other calls. Internally, it will make a call like `require('blink.cmp').setup(your_opts)`.
> 
> ```lua
> {
>   'saghen/blink.cmp',
>   lazy = false,
>   dependencies = 'rafamadriz/friendly-snippets',
>   version = 'v0.*',
>   opts = {
>     -- 'default' for mappings similar to built-in completion
>     -- 'super-tab' for mappings similar to vscode (tab to accept, arrow keys to navigate)
>     -- 'enter' for mappings similar to 'super-tab' but with 'enter' to accept
>     -- see the "default configuration" section below for full documentation on how to define
>     -- your own keymap.
>     keymap = { preset = 'default' },
> 
>     appearance = {
>       -- Sets the fallback highlight groups to nvim-cmp's highlight groups
>       -- Useful for when your theme doesn't support blink.cmp
>       -- will be removed in a future release
>       use_nvim_cmp_as_default = true,
>       -- Set to 'mono' for 'Nerd Font Mono' or 'normal' for 'Nerd Font'
>       -- Adjusts spacing to ensure icons are aligned
>       nerd_font_variant = 'mono'
>     },
> 
>     -- default list of enabled providers defined so that you can extend it
>     -- elsewhere in your config, without redefining it, via `opts_extend`
>     sources = {
>       completion = {
>         enabled_providers = { 'lsp', 'path', 'snippets', 'buffer' },
>       },
>     },
> 
>     -- experimental auto-brackets support
>     -- completion = { accept = { auto_brackets = { enabled = true } } }
> 
>     -- experimental signature help support
>     -- signature = { enabled = true }
>   },
>   -- allows extending the enabled_providers array elsewhere in your config
>   -- without having to redefine it
>   opts_extend = { "sources.completion.enabled_providers" },
> }
> ```

I did as you advised anyway the download error

**Reply by saghen** [reply, marked as answer]
What version of neovim are you on? It seems like you might be on a version <0.10 but the plugin requires 0.10+

**Reply by DartMitai** [reply]
my version `NVIM v0.9.5`

**Reply by DartMitai** [reply]
I raised the version, the error is gone, but the plugin does not want to work, this is another problem
